File: app/src/main/java/co/lateralview/myapp/domain/repository/interfaces/SharedPreferencesManager.java
package co.lateralview.myapp.domain.repository.interfaces;

import java.lang.reflect.Type;

public interface SharedPreferencesManager {

    boolean saveBlocking(String key, String value);

    String getString(String key);

    String getString(String key, String defaultValue);

    int getInt(String key, int defaultValue);

    <T> T get(String key, Class<T> type);

    <T> T get(String key, Type type);

    void clear();
}

File: app/src/main/java/co/lateralview/myapp/domain/repository/interfaces/SharedPreferencesManager.kt
package co.lateralview.myapp.domain.repository.interfaces

import java.lang.reflect.Type

interface SharedPreferencesManager {

    fun saveBlocking(key: String, value: String): Boolean

    fun getString(key: String): String

    fun getString(key: String, defaultValue: String): String

    fun getInt(key: String, defaultValue: Int): Int

    operator fun <T> get(key: String, type: Class<T>): T?

    operator fun <T> get(key: String, type: Type): T?

    fun clear()
}